### PR TITLE
fix(profile): set page_owner_guid if no username is set

### PR DESCRIPTION
**Pull request by jdalsem**

### CODE REVIEW

**Review by jeabakker — APPROVED** [review]

File: mod/profile/views/default/resources/profile/view.php [hunk 1]
```diff
@@ -7,7 +7,11 @@
 	$user = elgg_get_logged_in_user_entity();
 }
 
-elgg_entity_gatekeeper($user->guid, 'user');
```
Comment (hypeJunction) [inline mod/profile/views/default/resources/profile/view.php]: Keep the gatekeeper

Comment (jeabakker) [reply mod/profile/views/default/resources/profile/view.php]: `$user` could be empty (invalid username) and then you get a notice about getting a property on a non object

Comment (hypeJunction) [reply mod/profile/views/default/resources/profile/view.php]: You can throw, but keep the gatekeeper.
I am thinking about GDPR and all the ways we will need to be restricting access to profiles, so keep the gatekeeper in for other purposes.

Comment (jdalsem) [reply mod/profile/views/default/resources/profile/view.php]: i restored it

**Pull request merged**